livekit-plugins/livekit-plugins-deepgram/livekit/plugins/deepgram/tts.py

```
@@ -0,0 +1,341 @@
+                        self._segments_ch.send_nowait(word_stream)
+                    word_stream.push_text(input)
+                elif isinstance(input, self._FlushSentinel):
+                    if word_stream:
+                        word_stream.end_input()
+                    word_stream = None
+            self._segments_ch.close()
+
+        @utils.log_exceptions(logger=logger)
+        async def _run_segments(ws: aiohttp.ClientWebSocketResponse):
+            async for word_stream in self._segments_ch:
+                async for word in word_stream:
+                    speak_msg = {"type": "Speak", "text": f"{word.token} "}
+                    await ws.send_str(json.dumps(speak_msg))
+
+                # Always flush after a segment
+                flush_msg = {"type": "Flush"}
+                await ws.send_str(json.dumps(flush_msg))
+
+            # after all segments, close
+            close_msg = {"type": "Close"}
+            await ws.send_str(json.dumps(close_msg))
+
+        async def recv_task(ws: aiohttp.ClientWebSocketResponse):
+            last_frame: rtc.AudioFrame | None = None
+
+            def _send_last_frame(*, segment_id: str, is_final: bool) -> None:
+                nonlocal last_frame
+                if last_frame is not None:
+                    self._event_ch.send_nowait(
+                        tts.SynthesizedAudio(
+                            request_id=request_id,
+                            segment_id=segment_id,
+                            frame=last_frame,
+                            is_final=is_final,
+                        )
+                    )
+                    last_frame = None
+
+            while True:
+                msg = await ws.receive()
+                if msg.type in (
+                    aiohttp.WSMsgType.CLOSE,
+                    aiohttp.WSMsgType.CLOSED,
+                    aiohttp.WSMsgType.CLOSING,
+                ):
+                    return
+
```

Errors are not propagated to the _run_task. So no retry will happen on failure